### PR TITLE
Draft order shipping method doesn't update after removed all products.

### PR DESCRIPTION
**Pull request by dexon44**
I want to merge this change because it fix bug with OrderLineDelete mutation doesn't remove shipping method and recalculate total.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [x] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/order/bulk_mutations/draft_orders.py [hunk 1]
```diff
@@ -34,14 +34,14 @@ def clean_instance(cls, info, instance):
             )
 
 
-class DraftOrderLinesBulkDelete(ModelBulkDeleteMutation):
+class DeprecatedDraftOrderLinesBulkDelete(ModelBulkDeleteMutation):
```
Comment (IKarbowiak) [inline saleor/graphql/order/bulk_mutations/draft_orders.py:37]: We do not change the name of the deprecated mutation.

File: saleor/graphql/order/bulk_mutations/draft_orders.py [hunk 2]
```diff
@@ -34,14 +34,14 @@ def clean_instance(cls, info, instance):
...
     class Arguments:
         ids = graphene.List(
             graphene.ID, required=True, description="List of order lines IDs to delete."
         )
 
     class Meta:
-        description = "Deletes order lines."
+        description = "DEPRECATED: Will be removed in Saleor 4.0."
```
Comment (IKarbowiak) [inline saleor/graphql/order/bulk_mutations/draft_orders.py:44]: The description of the mutation should be left.
```suggestion
        description = "Deletes order lines. DEPRECATED: Will be removed in Saleor 4.0."
```

**Comment by patrys** [comment]
Task linked: [SALEOR-2502 Draft order: Shipping method doesn't update after removed all products](https://app.clickup.com/t/2549495/SALEOR-2502)

File: saleor/graphql/order/mutations/orders.py [hunk 3]
```diff
@@ -824,6 +854,28 @@ def perform_mutation(cls, _root, info, id):
...
+            )
+            recalculate_order(order)
+            return OrderLineDelete(order=order, order_line=line)
+
+        order.shipping_method = None
+        order.shipping_price = zero_taxed_money(order.currency)
+        order.shipping_method_name = None
+        order.save(
+            update_fields=[
+                "currency",
```
Comment (fowczarek) [inline saleor/graphql/order/mutations/orders.py:872]: Why do we need to update order currency?

**Review by fowczarek — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/order/mutations/orders.py [hunk 4]
```diff
@@ -824,6 +854,28 @@ def perform_mutation(cls, _root, info, id):
...
         line.id = db_id
 
+        if order.is_shipping_required():
+            events.order_removed_products_event(
+                order=order,
+                user=info.context.user,
+                app=info.context.app,
+                order_lines=[(line.quantity, line)],
+            )
+            recalculate_order(order)
```
Comment (maarcingebala) [inline saleor/graphql/order/mutations/orders.py:864]: Now, this piece of code is duplicated - we call it when shipping is required, but also when it's not, we call it at the end of the mutation. We could easily remove the duplication:
- invert the if condition to `if not order.is_shipping_required()`
- inside the `if` block - clear the shipping method (the logic that is done in lines 867-878)
- after exiting the `if` block, create the history event and recalculate order

File: saleor/graphql/order/mutations/orders.py [hunk 5]
```diff
@@ -220,7 +220,9 @@ def save(cls, info, instance, cleaned_input):
 
 class OrderUpdateShippingInput(graphene.InputObjectType):
     shipping_method = graphene.ID(
-        description="ID of the selected shipping method.", name="shippingMethod"
+        description="ID of the selected shipping method,"
+        " pass null to remove currently assigned shippingMethod.",
```
Comment (maarcingebala) [inline saleor/graphql/order/mutations/orders.py:224]:
```suggestion
        " pass null to remove currently assigned shipping method.",
```

File: saleor/graphql/order/mutations/orders.py [hunk 6]
```diff
@@ -251,11 +253,16 @@ class Arguments:
...
-            description="Fields required to change shipping method of the order."
+            description="Fields required to change shipping method of the order.",
+            required=True,
         )
 
     class Meta:
-        description = "Updates a shipping method of the order."
+        description = (
+            "Updates a shipping method of the order."
+            " Requires shipping Id to update, when null is passed "
```
Comment (maarcingebala) [inline saleor/graphql/order/mutations/orders.py:263]:
```suggestion
            " Requires shipping method ID to update, when null is passed "
```

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**